apps/prairielearn/elements/pl-big-o-input/pl-big-o-input.mustache

```diff
@@ -61,7 +61,7 @@
                     </div>`,
             });
             $('#big-o-input-submission-{{uuid}} [data-toggle="popover"]').on('shown.bs.popover', function() {
-                MathJax.typesetPromise();
+                MathJax.startup.promise.then(() => MathJax.typesetPromise());
```

Comment:
I grepped across all questions and unfortunately, there are course elements that make direct use of `MathJax.typesetPromise()`, which means we'll have to keep that as an API to avoid breaking them.

Reply:
> I grepped across all questions and unfortunately, there are course elements that make direct use of `MathJax.typesetPromise()`, which means we'll have to keep that as an API to avoid breaking them.

Keeping `MathJax.typesetPromise` is not a problem in most cases, it would only cause a problem if it is called before Mathjax finishes the initialization. That said, I could add a new fake `typesetPromise` function that does an appropriate action if Mathjax is not yet loaded, for additional safety. Thoughts? Something similar to what was done with `Mathjax.startup.promise`.

Reply:
Done in 13862461454694f3157a70428e4b62ccc6ca0b02. In this case, a call to `MathJax.typesetPromise()` directly will work either before or after MathJax is finished loading.